### ISSUE TITLE
`no_effect` creates duplicates reports with `rustc`'s `path_statements` lints

### ISSUE THREAD

**Issue by sosthene-nitrokey** [rust-lang/rust-clippy#11547]
### Summary

When line triggers rustc's `path_statements` lints, it also triggers clippy's `no_effect` lint. This leads to confusing duplicate warnings when running clippy, because to distinct lints warn of the same thing.

I believe having both show at the same time is not intended. Maybe the `no_effect` warning in this specific case could be skipped if it's not configured to `deny`?

### Reproducer

I tried this code:

```rust
#[derive(Default)]
pub struct A;

impl A {
    const CONSTANT: () = ();
    pub fn new() -> A {
        Self::CONSTANT;
        A
    }
}
```

I expected to see this happen:

```
warning: path statement with no effect
 --> src/lib.rs:7:9
  |
7 |         Self::CONSTANT;
  |         ^^^^^^^^^^^^^^^
  |
  = note: `#[warn(path_statements)]` on by default

warning: `playground` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.99s
```

or 

```
   Checking playground v0.0.1 (/playground)
warning: statement with no effect
 --> src/lib.rs:7:9
  |
7 |         Self::CONSTANT;
  |         ^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#no_effect
  = note: `#[warn(clippy::no_effect)]` on by default
warning: `playground` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.99s
```

Instead, both warning messages appear at the same time:

```
    Checking playground v0.0.1 (/playground)
warning: statement with no effect
 --> src/lib.rs:7:9
  |
7 |         Self::CONSTANT;
  |         ^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#no_effect
  = note: `#[warn(clippy::no_effect)]` on by default

warning: path statement with no effect
 --> src/lib.rs:7:9
  |
7 |         Self::CONSTANT;
  |         ^^^^^^^^^^^^^^^
  |
  = note: `#[warn(path_statements)]` on by default

warning: `playground` (lib) generated 2 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.99s
```

### Version

```text
Build using the Nightly version: 1.74.0-nightly

(2023-09-20 3223b0b5e8dadda3f76c)
```
